activerecord/lib/active_record/relation.rb

```diff
@@ -627,6 +639,28 @@ def delete_by(*args)
       where(*args).delete_all
     end
 
+    # Schedule the query to be performed from a background thread pool.
+    #
+    #   Post.where(published: true).defer # => #<ActiveRecord::Relation>
+    def defer
```

Comment:
To me at least, `defer` means that something will be loaded in the main thread, but at a later time (ie. lazy).

How about `load_async`?

Reply:
As with the previous PR I'm totally open to naming suggestions.

I used `defer` because it's a quite common term in various async frameworks & such. The intent is not to say that the query is deferred, but that the use of it's result is. So maybe `deferred` would be more correct.

But yeah, `load_async` would probably speak more to most users.

Reply:
Done.